data/registry_model/registry_proxy_model.py

```diff
@@ -185,13 +191,36 @@ def lookup_manifest_by_digest(
                 oci.tag.set_tag_end_ms(db_tag, new_expiration)
                 # if the manifest is a child of a manifest list in this repo, renew
                 # the parent(s) manifest list tag too.
-                links = ManifestChild.select(ManifestChild.manifest_id).where(
-                    (ManifestChild.repository_id == repository_ref.id)
-                    & (ManifestChild.child_manifest_id == wrapped_manifest.id)
+                # select tag ids by most recent lifetime_end_ms uniquely by name,
+                # based on the link between sub-manifest and parent manifest in the
+                # manifest link.
+                subq = (
+                    TagTable.select(
+                        fn.row_number()
+                        .over(partition_by=TagTable.name, order_by=TagTable.lifetime_end_ms.desc())
+                        .alias("rnk"),
+                        TagTable.id,
+                        TagTable.name,
+                        TagTable.lifetime_end_ms,
+                    )
+                    .join(ManifestChild, on=(TagTable.manifest_id == ManifestChild.manifest_id))
+                    .where(
+                        (ManifestChild.repository_id == repository_ref.id)
+                        & (ManifestChild.child_manifest_id == wrapped_manifest.id)
+                    )
+                    .order_by(TagTable.name)
+                    .order_by_extend(TagTable.lifetime_end_ms.desc())
                 )
-                parent_ids = [link.manifest_id for link in links]
+                items = (
```

Comment:
Will this query use indexes?

Reply:
I'm not sure about the order by name tbh, but that's there for troubleshooting purposes so I'll remove it. the rest should be using the indexes.